### ISSUE TITLE
fle_utils.videos.download_video fails for some videos

### ISSUE THREAD

**Issue by benjaoming** [learningequality/ka-lite#4782]
The method of using `response.type.startswith` is broken. Example for a working video:

    import urllib
    __, response = urllib.urlretrieve("http://kalite.learningequality.org/download/videos/leDYnoNSvD4.mp4/leDYnoNSvD4.mp4", "/tmp/test1.mp4")
    if not response.type.startswith("video"):
        raise URLNotFound("Video was not found!")
    else:
        print("This works")

This one will output "This works".

Example of a video which this method does not function with:

    import urllib
    __, response = urllib.urlretrieve("http://kalite.learningequality.org/download/videos/C6PUlTYnxLY.mp4/C6PUlTYnxLY.mp4", "/tmp/test2.mp4")
    if not response.type.startswith("video"):
        raise URLNotFound("Video was not found!")
    else:
        print("This works")

This one will raise URLNotFound.

**Comment by benjaoming** [comment]
This has probably been broken for a long time, as it also affects the 0.15 videos.

Once again, the whole video download part should be completely rewritten :) I think the time it takes to rewrite it in a properly functioning way is better than sorting out all the bugs it causes. Maybe rewriting it can serve as a kind of sketching for Kolibri's download manager?

**Comment by benjaoming** [comment]
```
>>> __, response = urllib.urlretrieve("http://kalite.learningequality.org/download/videos/C6PUlTYnxLY.mp4/C6PUlTYnxLY.mp4", "/tmp/test2.mp4")
>>> print response
x-amz-id-2: JYTqwM280uqO+GUqrEcOYVe4d6bdVkK1+nbIKRU4+DSyhWUZ6x12CL6bAUrNCFoZGjQIJIpceRU=
x-amz-request-id: 952CA6C1F9F48A87
Date: Wed, 13 Jan 2016 12:16:55 GMT
Last-Modified: Sat, 27 Apr 2013 00:27:29 GMT
ETag: "5dc8676554baf80b5f575b847d2a64b3-2"
Accept-Ranges: bytes
Content-Type: 
Content-Length: 127454923
Connection: keep-alive
Server: AmazonS3

>>> print dir(response)
['__contains__', '__delitem__', '__doc__', '__getitem__', '__init__', '__iter__', '__len__', '__module__', '__setitem__', '__str__', 'addcontinue', 'addheader', 'dict', 'encodingheader', 'fp', 'get', 'getaddr', 'getaddrlist', 'getallmatchingheaders', 'getdate', 'getdate_tz', 'getencoding', 'getfirstmatchingheader', 'getheader', 'getheaders', 'getmaintype', 'getparam', 'getparamnames', 'getplist', 'getrawheader', 'getsubtype', 'gettype', 'has_key', 'headers', 'iscomment', 'isheader', 'islast', 'items', 'keys', 'maintype', 'parseplist', 'parsetype', 'plist', 'plisttext', 'readheaders', 'rewindbody', 'seekable', 'setdefault', 'startofbody', 'startofheaders', 'status', 'subtype', 'type', 'typeheader', 'unixfrom', 'values']
```

response.status is unfortunately None
response.type is an empty string

Proposal:

We can use `response.Content-Length` to match the downloaded file. Example:

    len(open("/tmp/test2.mp4", "rb").read())
    127454923
    response["Content-Length"]
    '127454923'

    >>> str(len(open("/tmp/test2.mp4", "rb").read())) == response["Content-Length"]
    True

**Comment by benjaoming** [comment]
Also, since we already depend on `requests`, we should definitely use this lib instead of urllib, this is also recommended in the official Python docs.

**Comment by aronasorman** [comment]
> Also, since we already depend on requests

The version bundled in ka-lite doesn't support streaming requests I believe, so that's a prerequisite (0.17?)

**Comment by benjaoming** [comment]
Argh, okay, let's get the latest requests in 0.17, will do that while unbundling stuff.

But as for the Content-Length thing, Kiwix people are testing it out by downloading ALL videos so we can be pretty certain if this approach will work or not :)

**Comment by benjaoming** [comment]
The reportback is that all videos succeeded downloading using the Content-Length header comparison. So I'd propose switching to that.

**Comment by rtibbles** [comment]
Fixed in #4784
